### PR TITLE
fix getting sizes for DO

### LINKED ISSUE #4214: Fix getting sizes for DO during adding new ND
**User Story**
Get DO token from secret. Right now it takes token from cloud spec

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: fix getting sizes for DO

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4214

```release-note
NONE
```

### CODE REVIEW

**Review by nikhita — APPROVED** [review]
/lgtm
/approve

**Comment by alvaroaleman** [comment]
This breaks the endpoint for all clusters that do not have a secret yet, which is any cluster that was created before Nikhitas change, as we don't yet have a migration.

This must use the same Code the provider itself uses to get the token and not  duplicate anything. This must also be able to cope with both the secretRef and the token being directly on the cluster CR:

**Comment by alvaroaleman** [comment]
Can we please just revert https://github.com/kubermatic/kubermatic/pull/3946 instead?

**Comment by zreigz** [comment]
what about something like this:
```
		doToken := cluster.Spec.Cloud.Digitalocean.Token
		if len(doToken) == 0 {
			secret, err := kubernetesprovider.GetCredentialSecret(ctx, privilegedClusterProvider.GetSeedClusterAdminRuntimeClient(), cluster.Spec.Cloud.Digitalocean.CredentialsReference.Name)
			if err != nil {
				return nil, common.KubernetesErrorToHTTPError(err)
			}

			doTokenBytes, ok := secret.Data[resources.DigitaloceanToken]
			if !ok {
				return nil, errors.NewNotFound("token for ", req.ClusterID)
			}
			doToken = string(doTokenBytes)
		}

		return digitaloceanSize(ctx, doToken)
```

**Comment by zreigz** [comment]
it will make the endpoint backward compatible

**Comment by alvaroaleman** [comment]
There is `resources.GetDigitalOceanCredentials` for this. Another problem I found is that the `Validate` of the DO provider just does a
```
static := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: spec.Digitalocean.Token})
```

I am not sure why this works (probably because its called before we unset the Token on the cluster?) but it needs to be fixed.

**Comment by zreigz** [comment]
> There is `resources.GetDigitalOceanCredentials` for this. Another problem I found is that the `Validate` of the DO provider just does a
> 
> ```
> static := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: spec.Digitalocean.Token})
> ```
> 
> I am not sure why this works (probably because its called before we unset the Token on the cluster?) but it needs to be fixed.

This part is called before secret creation and cluster creation when cloud spec still contains credentials fields, so it's perfectly fine

File: api/pkg/provider/kubernetes/credentials.go [hunk 1]
```diff
@@ -12,6 +12,14 @@ import (
 	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+func GetCredentialSecret(ctx context.Context, seedClient ctrlruntimeclient.Client, secretName string) (*corev1.Secret, error) {
```
Comment (alvaroaleman) [inline api/pkg/provider/kubernetes/credentials.go]: Is this a leftover? I can't find it referenced anywhere

**Pull request merged**